3696/1/top_gems.rb

```diff
@@ -0,0 +1,50 @@
+require 'yaml'
+require 'nokogiri'
+require 'open-uri'
+require 'optparse'
+require './presenter'
+require './scraper'
+require './name_to_link_transformer.rb'
+
+class TopGems
+  def ranking(file, name)
+    @ranking ||= file.each_with_object({}) do |str, hsh|
+      next unless str.start_with? name
+
+      @transformer = NameToLinkTransformer.new(str)
+
+      hsh[str] = Scraper.new(@transformer.link) if @transformer.valid
+    end
+  end
+
+  def run
+    file, name, top = options
```

Comment:
now it's harder to change arguments order or to add new one. You must copy this code everywhere you need it. 
Try to use hashes instead